scripts/migrate_maas_instance_ids.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+set -e
+
+#This script script is intended to run only against controller machines that
+#have upgraded from a MAAS 1.9 provider to a MAAS 2.x provider. Do not use this
+#script otherwise. The javascript below is horrible. It is designed to run
+#operations in bulk targeting a minimum version of mongo 3.2 which has little
+#functionality in the way of aggregation.
+
+usage() { echo "usage: $0 [-rh]" 1>&2; exit 1; }
+
+while getopts 'rh' flag; do
+    case "${flag}" in
+        r) rollback_flag=true
+           ;;
+        h) usage
+           ;;
+        *) usage
+           ;;
+    esac
+done
+
+if [ $rollback_flag ]
+then
+    #This script warps a MAAS 1.9 style URL around the instance_id. You can run
+    #this script with the -r flag for convenience but you should really just
+    #reload the backup that you have created with mongoexport or otherwise if
+    #you run into trouble.
+    mongo_script=$(cat <<'JAVASCRIPT'
+db.instanceData.aggregate([{
```

Comment:
I find this easier to read formatted slightly differently:
```
db.instanceData.aggregate([{
    "$project": {
        instanceid: {
            "$cond": {
                if: {
                    "$eq": [{"$substr": ["$instanceid", 0, 20]}, "/MAAS/api/1.0/nodes/"]
                },
                then: {
                    "$substr": ["$instanceid", 20, -1]
                },
                else: "$instanceid"
            }
        },
        document: "$$ROOT"
    }
}, {
    "$out": "tmp"
}]);
ops = db.instanceData.find().map(function(item) {
    newid = db.tmp.findOne({
        "document.instanceid": item.instanceid
    });
    if (item.instanceid.slice(-1) === "/") {
        return {
            "updateOne": {
                "filter": {
                    instanceid: newid.document.instanceid
                },
                "update": {
                    "$set": {
                        instanceid: newid.instanceid.slice(0, -1)
                    }
                }
            }
        }
    } else {
        return {}
    }
});
db.instanceData.bulkWrite(ops);
```

Mostly not having 
```
}
, more
```
and making sure things like "$ne": [{}, VALUE]
end up on one line, rather than breaking it in odd places.